cmd/snap/inhibit.go

```diff
@@ -25,12 +25,35 @@ import (
 	"os"
 	"time"
 
+	"github.com/godbus/dbus"
 	"github.com/snapcore/snapd/cmd/snaplock/runinhibit"
+	"github.com/snapcore/snapd/dbusutil"
 	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/logger"
 	"github.com/snapcore/snapd/progress"
 	"github.com/snapcore/snapd/usersession/client"
 )
 
+func tryNotifyRefreshToSnapDesktopIntegration(snapName string) (bool, error) {
+	// Check if Snapd-Desktop-Integration is available
+	conn, err := dbusutil.SessionBus()
+	if err != nil {
+		logger.Noticef("unable to connect dbus session: %v", err)
+		return false, nil
+	}
+	obj := conn.Object("io.snapcraft.SnapDesktopIntegration", "/io/snapcraft/SnapDesktopIntegration")
+	extraParams := make(map[string]dbus.Variant)
+	err = obj.Call("io.snapcraft.SnapDesktopIntegration.ApplicationIsBeingRefreshed", 0, snapName, runinhibit.HintFile(snapName), extraParams).Store()
```

Comment:
are you sending the HintFile because the desktop integration is monitoring it on its own?

Reply:
Yes. That way, SnapdDesktopIntegration can detect by itself when the snap is ready and close the notification. Anyway, the DBus API used also allows to not rely on that (just by passing an empty string as the lock file path), and close the notification programatically using another DBus method (I use that in the "refresh-running-snap-on-close" MR).